### ISSUE TITLE
media_manager errors out if file has been previously deleted

### ISSUE THREAD

**Issue by cyturralde** [blacktwin/JBOPS#273]
I am on Windows 10. If I run the media_manager.py with an action of "show", it shows the shows/movies that were watched. If I run the same command but with "delete" action instead I get an error.

```
The following items were added before None and marked for deletion.
Traceback (most recent call last):
  File "F:\OneDrive\Desktop\remove-plex.py", line 624, in <module>
    plex_deletion(watched_by_all, libraries, opts.toggleDeletion)
  File "F:\OneDrive\Desktop\remove-plex.py", line 296, in plex_deletion
    plex_item = plex.fetchItem(item)
  File "C:\Python27\lib\site-packages\plexapi\base.py", line 136, in fetchItem
    for elem in self._server.query(ekey):
  File "C:\Python27\lib\site-packages\plexapi\server.py", line 381, in query
    raise BadRequest('(%s) %s; %s %s' % (response.status_code, codename, response.url, errtext))
plexapi.exceptions.BadRequest: (404) not_found; http://localhost:32400/library/metadata/12290 <html><head><title>Not Found</title></head><body><h1>404 Not Found</h1></body></html>
```

This happens in multiple folders I am just using this one as an example. Based on what I can tell, when it runs into a show or movie that has been watched but already deleted, either manually or via another program, it errors out, rather than skipping it and moving to the next.

**Comment by blacktwin** [comment]
I'll be using PR #270 to add changes to address this issue. Thanks for letting me know. I'll call out this issue in the commit when its addressed and added.

**Comment by blacktwin** [comment]
@cyturralde Can you try the PR I mentioned above, ?I've added a potential fix for your issue.

**Comment by cyturralde** [comment]
@blacktwin Sorry for the delay I didn't get a notification there was a response for some reason. Here is the error I get now when I run with the delete command: 

Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
The following items were added before None and marked for deletion.
Traceback (most recent call last):
  File "F:\OneDrive\Desktop\remove-plex.py", line 695, in <module>
    plex_deletion(watched_by_all, libraries, opts.toggleDeletion)
  File "F:\OneDrive\Desktop\remove-plex.py", line 303, in plex_deletion
    plex_item = plex.fetchItem(int(item.rating_key))
AttributeError: 'unicode' object has no attribute 'rating_key'



This section: 

Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None
Tautulli API cmd 'get_metadata' failed: None

also shows up if I run the show command.

**Comment by blacktwin** [comment]
I don't think you pulled the file from the PR. Use this [link](https://github.com/blacktwin/JBOPS/blob/3421661e222ab97b3355c48b004298ae4c35e0fa/utility/media_manager.py) to update your script.

**Comment by cyturralde** [comment]
I just copied and pasted from that link and ran this command:

remove-plex.py  --libraries "*library*" --select watched --users *user* --action delete

I got this error:

Finding watched items from user: *user*
Checking library: '*library*' watch statuses...
The following items were added before None and marked for deletion.
Traceback (most recent call last):
  File "F:\OneDrive\Desktop\remove-plex.py", line 706, in <module>
    plex_deletion(watched_by_all, libraries, opts.toggleDeletion)
  File "F:\OneDrive\Desktop\remove-plex.py", line 309, in plex_deletion
    plex_item = plex.fetchItem(int(item.rating_key))
AttributeError: 'unicode' object has no attribute 'rating_key'

**Comment by blacktwin** [comment]
Updated again adding the additional exception for `AttributeError`. You can find it [here](https://github.com/blacktwin/JBOPS/blob/592029cee393e47336d76eb56f152f6b191600a4/utility/media_manager.py).

**Comment by cyturralde** [comment]
No more error but here is what is happening now: 

python remove-plex.py --libraries "*library*" --select watched --users *user*--action delete
Finding watched items from user: *user*
Checking library: '*library*' watch statuses...
The following items were added before None and marked for deletion.
Item: 11705 may already have been deleted.
Item: 12285 may already have been deleted.
Item: 12319 may already have been deleted.
Item: 12146 may already have been deleted.
Item: 12142 may already have been deleted.
Item: 11680 may already have been deleted.
Item: 12140 may already have been deleted.
Emptying Trash from library Mom


The problem is those items it are listing have not been deleted already. They are still there. It shows them as "watched" if I do --action show.

**Comment by blacktwin** [comment]
>The problem is those items it are listing have not been deleted already. They are still there. It shows them as "watched" if I do --action show.

Not deleted in Plex or its library? The watched information is coming from Tautulli so it would still report as being watched.

**Comment by cyturralde** [comment]
Its not deleted in plex. The show is still there.

**Comment by cyturralde** [comment]
Just for testing purposes I tried a different user and folder and got the same issue.

**Comment by cyturralde** [comment]
I noticed in Tautulli an error showing up:


2021-01-06 15:54:05 | DEBUG | Tautulli Pmsconnect :: No library data.
-- | -- | --


I wonder if that is related?

**Comment by blacktwin** [comment]
Another [update](https://github.com/blacktwin/JBOPS/blob/0aa5abd6c8ebdc01ea15ba34720f460cbf695e5f/utility/media_manager.py). 

>I wonder if that is related?

Probably not good or helpful.

**Comment by cyturralde** [comment]
That seems to have done the trick. Just for my understanding, based on the latest change, for some reason it was fetching as a string instead of integer, so you just had to convert it, in an if statement?

**Comment by blacktwin** [comment]
Yes. I didn't do any kind of deep dive to check as to why. I'd need to spin up my test server to prep to actually test a lot of conditions and scenarios.

**Comment by cyturralde** [comment]
Makes since. I was just trying to understand it. Thanks for the help.

**Comment by blacktwin** [comment]
This issue should now be resolved via #270
